runtime/src/accounts.rs

```diff
@@ -931,6 +931,26 @@ impl Accounts {
             .collect()
     }
 
+    #[allow(clippy::needless_collect)]
+    pub fn lock_accounts_with_status<'a>(
+        &self,
+        txs: impl Iterator<Item = &'a SanitizedTransaction>,
+        status: impl Iterator<Item = &'a Result<()>>,
```

Comment:
```suggestion
    pub fn lock_accounts_with_results<'a>(
        &self,
        txs: impl Iterator<Item = &'a SanitizedTransaction>,
        results: impl Iterator<Item = &'a Result<()>>,
```

Reply:
Wdyt? If we stick with "status", it should probably be plural, and I don't care for the word "statuses"...